File: 2018/utils/data_reader.rb
module DataReader
  def read_data(file_path, file='input.txt')
    File.readlines("#{file_path}/#{file}")
  end
end

# USE CASE
=begin
require_relative('../../utils/data_reader.rb')
include DataReader

read_data(File.dirname(__FILE__))
=end

File: 2018/utils/dataReader.js
const fs = require("fs");

// Requires dir input from relative file to get the path.  Directory must have input.txt file
function dataReader(dir, file = "input.txt") {
  const contents = fs.readFileSync(`${dir}/${file}`, "utf8");
  return contents.split("\n");
}

module.exports = dataReader;

// Use Case
/*
const dataReader = require("../../utils/dataReader");
const input = dataReader(__dirname);
*/
